leaderelection panic when failing to renew lease

**What happened**:

Using leaderelection with ConfigMapLock, I get a panic when I fail to renew the lease.

```
2020-02-04T03:50:27.372845Z     info    failed to renew lease istio-system/istio-leader: timed out waiting for the condition
2020-02-04T03:50:27.372888Z     info    leader election lock lost
panic: assignment to entry in nil map

goroutine 44 [running]:
k8s.io/client-go/tools/leaderelection/resourcelock.(*ConfigMapLock).Update(0xc000a4a280, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        k8s.io/client-go@v0.17.2/tools/leaderelection/resourcelock/configmaplock.go:90 +0x129
k8s.io/client-go/tools/leaderelection.(*LeaderElector).release(0xc00012d7a0, 0x1bf08eb00)
        k8s.io/client-go@v0.17.2/tools/leaderelection/leaderelection.go:306 +0xc9
k8s.io/client-go/tools/leaderelection.(*LeaderElector).renew(0xc00012d7a0, 0x2a49ec0, 0xc00119c180)
        k8s.io/client-go@v0.17.2/tools/leaderelection/leaderelection.go:294 +0x166
k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run(0xc00012d7a0, 0x2a49ec0, 0xc00119c140)
        k8s.io/client-go@v0.17.2/tools/leaderelection/leaderelection.go:208 +0x15d
istio.io/istio/pilot/pkg/leaderelection.(*LeaderElection).Run.func1(0xc00012d7a0, 0x2a49ec0, 0xc000e1ba00, 0xc000ac6c50)
        istio.io/istio@/pilot/pkg/leaderelection/leaderelection.go:54 +0x3f
created by istio.io/istio/pilot/pkg/leaderelection.(*LeaderElection).Run
        istio.io/istio@/pilot/pkg/leaderelection/leaderelection.go:53 +0x124
```


**What you expected to happen**: 

No crash


**How to reproduce it (as minimally and precisely as possible)**:

* Create a leaderelection setup with ReleaseOnCancel=true
* Run the program, allow a lock to be acquired.
* Change RBAC permissions to block configmap `update` access
* Eventually we fail to acquire the lock and run into the panic above

--- a/staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go
+++ b/staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go
@@ -86,6 +86,9 @@ func (cml *ConfigMapLock) Update(ler LeaderElectionRecord) error {
 	if err != nil {
 		return err
 	}
+	if cml.cm.Annotations == nil {
+		cml.cm.Annotations = make(map[string]string)
+	}
 	cml.cm.Annotations[LeaderElectionRecordAnnotationKey] = string(recordBytes)
 	cml.cm, err = cml.Client.ConfigMaps(cml.ConfigMapMeta.Namespace).Update(cml.cm)
 	return err

